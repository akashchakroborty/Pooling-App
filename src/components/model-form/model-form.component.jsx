import React, { useState } from "react";
import { connect } from "react-redux";
import { withRouter } from "react-router-dom";
import PropTypes from "prop-types";

import QuestionsListActionTypes from "../../redux/questions-list/questions-list.types";

import { FormContainer, InputTextArea, InputText } from "./model-form.styles";

const ModelForm = ({ dispatch, history }) => {
  const [question, setQuestion] = useState({
    questionTitle: "",
    choices: ""
  });

  const { questionTitle, choices } = question;

  const getChoicesArray = textchoises => textchoises.trim().split(/\s*,\s*/);

  const handleChange = event => {
    event.preventDefault();
    const { value, name } = event.target;
    setQuestion({ ...question, [name]: value });
  };

  const handleSubmit = event => {
    event.preventDefault();
    dispatch({
      type: QuestionsListActionTypes.ADD_NEW_QUESTION_START,
      payload: {
        callback: () => {
          history.push(`/`);
        },
        questionData: {
          question: questionTitle,
          choices: getChoicesArray(choices)
        }
      }
    });
  };

  return (
    <div>
      <FormContainer onSubmit={handleSubmit}>
        <InputText
          type="text"
          name="questionTitle"
          placeholder="Type new question"
          value={questionTitle}
          onChange={handleChange}
          required
        />
        <InputTextArea
          type="textarea"
          name="choices"
          placeholder="List the choices followed by a coma. Example: A, B, C, D"
          value={choices}
          onChange={handleChange}
          required
        />
        <button type="submit">Submit</button>
      </FormContainer>
    </div>
  );
};

ModelForm.propTypes = {
  history: PropTypes.object.isRequired,
  dispatch: PropTypes.func.isRequired
};

export default withRouter(connect()(ModelForm));
