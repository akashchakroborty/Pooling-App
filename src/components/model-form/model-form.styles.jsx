import styled from "styled-components";

export const FormContainer = styled.form`
  display: flex;
  flex-direction: column;
  width: 80%;
  margin: 1rem auto;
`;

export const InputTextArea = styled.input`
  padding: 20px;
  margin: 1rem 0;
  resize: both;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  font-size: 16px;
`;

export const InputText = styled.input`
  padding: 20px;
  margin: 1rem 0;
  font-size: 16px;
`;
