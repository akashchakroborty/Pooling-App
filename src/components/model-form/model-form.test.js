import React from "react";
import { shallow } from "enzyme";
import { ModelForm } from "./model-form.component";
import { InputText, FormContainer } from "./model-form.styles";

describe("ModelForm component", () => {
  let mockHistory;
  let mockDispatch;
  let wrapper;

  beforeEach(() => {
    mockHistory = {
      push: jest.fn()
    };
    mockDispatch = jest.fn();

    const mockProps = {
      dispatch: mockDispatch,
      history: mockHistory
    };

    wrapper = shallow(<ModelForm {...mockProps} />);
  });

  it("should render ModelForm component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should call handleChange when value of InputText is changed", () => {
    const event = {
      preventDefault() {},
      target: {}
    };
    wrapper.find(InputText).simulate("change", event);
    expect.any(Function);
  });

  it("should call handleSubmit when FormContainer is Submited", () => {
    const event = {
      preventDefault() {}
    };
    wrapper.find(FormContainer).simulate("submit", event);
    expect.any(Function);
  });
});
