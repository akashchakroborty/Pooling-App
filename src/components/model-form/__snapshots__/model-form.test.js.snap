// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModelForm component should render ModelForm component 1`] = `ShallowWrapper {}`;
